### PR TITLE
[plugin.video.orftvthek] 0.11.0

### PR DESCRIPTION
**Pull request by s0faking**
### Description
compatible with kodi matrix
### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

File: plugin.video.orftvthek/resources/language/English/strings.po [hunk 1]
```diff
@@ -0,0 +1,234 @@
+# XBMC Media Center language file
```
Comment (enen92) [inline plugin.video.orftvthek/resources/language/English/strings.po]: change this to `plugin.video.orftvthek/resources/language/resource.language.en_gb/strings.po`

File: plugin.video.orftvthek/resources/language/German/strings.po [hunk 2]
```diff
@@ -0,0 +1,234 @@
+# XBMC Media Center language file
```
Comment (enen92) [inline plugin.video.orftvthek/resources/language/German/strings.po]: change this to `plugin.video.orftvthek/resources/language/resource.language.de_de/strings.po`

File: plugin.video.orftvthek/default.py [hunk 3]
```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/python
```
Comment (enen92) [inline plugin.video.orftvthek/default.py:1]: i'd be nice if you could turn the entrypoint as a simple dummy file which imports the main addon file and simply runs it:

```
from resources.lib import myaddon
myaddon.run()
```

Reasoning is that this is the only file that is not compiled to bytecode by the python interpreter

File: plugin.video.orftvthek/resources/lib/base.py [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+    else:
+        description = ""
+    play_all_item.setInfo(type="Video", infoLabels={"Title": name, "Plot": description})
+    xbmcplugin.addDirectoryItem(pluginhandle, play_all_url, play_all_item, isFolder=False, totalItems=-1)
+
+
+def createListItem(title, banner, description, duration, date, channel, videourl, playable, folder, backdrop, pluginhandle, subtitles=None, blacklist=False, contextMenuItems=None):
+    contextMenuItems = contextMenuItems or []
+
+    liz = xbmcgui.ListItem(title)
```
Comment (enen92) [inline plugin.video.orftvthek/resources/lib/base.py]: tip: adding `offscreen=True` whenever you create a `ListItem` will speed up your addon a bit

File: plugin.video.orftvthek/resources/lib/htmlscraper.py [hunk 5]
```diff
@@ -0,0 +1,1040 @@
...
+				channel = "LIVE"
+
+			uhd_streaming_url = self.getLivestreamUrl(data, 'uhdbrowser')
+			if uhd_streaming_url:
+				debugLog("Adding UHD Livestream")
+				uhdContextMenuItems = []
+				if inputstreamAdaptive and restart and online:
+					uhdContextMenuItems.append(('Restart', 'RunPlugin(plugin://%s/?mode=liveStreamRestart&link=%s)' % (
+						xbmcaddon.Addon().getAddonInfo('id'), link)))
+					uhd_final_title = "[Restart]%s[UHD] - %s%s" % (channel, title, time_str)
```
Comment (enen92) [inline plugin.video.orftvthek/resources/lib/htmlscraper.py]: this needs to be localized

File: plugin.video.orftvthek/resources/lib/htmlscraper.py [hunk 6]
```diff
@@ -0,0 +1,1040 @@
...
+				else:
+					uhd_final_title = "%s[UHD] - %s%s" % (channel, title, time_str)
+				self.html2ListItem(uhd_final_title, banner, "", state, time, channel, channel, generateAddonVideoUrl(uhd_streaming_url), None, False, True, uhdContextMenuItems)
+
+			streaming_url = self.getLivestreamUrl(data, self.videoQuality)
+			contextMenuItems = []
+			if inputstreamAdaptive and restart and online:
+				contextMenuItems.append(('Restart', 'RunPlugin(plugin://%s/?mode=liveStreamRestart&link=%s)' % (
```
Comment (enen92) [inline plugin.video.orftvthek/resources/lib/htmlscraper.py]: this too, it's shown in the UI

File: plugin.video.orftvthek/resources/lib/htmlscraper.py [hunk 7]
```diff
@@ -0,0 +1,1040 @@
...
+					xbmcaddon.Addon().getAddonInfo('id'), link)))
+				final_title = "[Restart]%s - %s%s" % (channel, title, time_str)
```
Comment (enen92) [inline plugin.video.orftvthek/resources/lib/htmlscraper.py]: same for this guy over here

File: plugin.video.orftvthek/resources/lib/htmlscraper.py [hunk 8]
```diff
@@ -0,0 +1,1040 @@
...
+			if len(section):
+				section = section[0]
+				streamingURL = 'https://playerapi-restarttv.ors.at/livestreams/%s/sections/%s/manifests/hls/?startTime=%s&X-Api-Key=%s' % (bitmovinStreamId, section.get('id'), section.get('metaData').get('timestamp'), ApiKey)
+
+				listItem = createListItem(title, image, description, duration, date, channel, streamingURL, True, False, self.defaultbackdrop, self.pluginhandle)
+				listItem.setProperty('inputstreamaddon', 'inputstream.adaptive')
+				listItem.setProperty('inputstream.adaptive.manifest_type', 'hls')
+				self.xbmc.Player().play(streamingURL, listItem)
+
+	@staticmethod
```
Comment (enen92) [inline plugin.video.orftvthek/resources/lib/htmlscraper.py]: it'd be nice if you can pass your addon through reindent: https://pypi.org/project/reindent/
Your identation in this file is a nightmare :) PEP recommends 4 spaces.

**Review by enen92 — CHANGES_REQUESTED** [review]
Thanks for submitting your addon to matrix. Unfortunately, at least the language folder structure has to be adjusted for travis to be happy. After that it's good to go in.
Other points are just suggestions, feel free to ignore or take them into consideration.
Cheers

File: plugin.video.orftvthek/resources/language/English/strings.po (hunk 1 above)
Comment (s0faking) [reply plugin.video.orftvthek/resources/language/English/strings.po]: done with the new commit

File: plugin.video.orftvthek/resources/language/German/strings.po (hunk 2 above)
Comment (s0faking) [reply plugin.video.orftvthek/resources/language/German/strings.po]: also done

File: plugin.video.orftvthek/default.py (hunk 3 above)
Comment (s0faking) [reply plugin.video.orftvthek/default.py:1]: that totally makes sense. fixed in the new commit

File: plugin.video.orftvthek/resources/lib/base.py (hunk 4 above)
Comment (s0faking) [reply plugin.video.orftvthek/resources/lib/base.py]: nice, thx for the hint. i added this on all ListItem calls. couldnt find anything useful about this in the docs. what does it do?

File: plugin.video.orftvthek/resources/lib/htmlscraper.py (hunk 5 above)
Comment (s0faking) [reply plugin.video.orftvthek/resources/lib/htmlscraper.py]: done. although the translation for the string is the same in german. the feature on the website/app is also called "Restart" in german.

File: plugin.video.orftvthek/resources/lib/htmlscraper.py (hunk 8 above)
Comment (s0faking) [reply plugin.video.orftvthek/resources/lib/htmlscraper.py]: sorry for the mess. i ran reindent on all py files.

File: plugin.video.orftvthek/resources/lib/base.py (hunk 4 above)
Comment (enen92) [reply plugin.video.orftvthek/resources/lib/base.py]: It avoids a GUI lock. Afaik this was not set as default due to compatibility reasons, long long ago.

**Review by enen92 — APPROVED** [review]

**Pull request merged**